static/js/activity.js

```diff
@@ -348,9 +354,17 @@ exports.build_user_sidebar = function () {
         return;
     }
 
-    var user_ids = filter_and_sort(presence.get_user_ids());
+    var user_ids;
+
+    if (meta.$user_list_filter.val().length) {
+        user_ids = filter_and_sort(presence.get_user_ids());
+    } else {
+        user_ids = filter_and_sort(people.get_all_persons().map(function (person) {
```

Comment:
This should be `people.get_realm_persons()`, not `people.get_all_persons`; the different is some cross-realm bots and things, that we happen to filter above but might as well be excluded from the beginning.

Reply:
Ah, okay so changing to this may mean I won't have to filter as I am below.

Reply:
Cool, let me know when updated.